Use feed level author if entry does not have one

der-lyse opened the pull request:
This merge request only addresses the feed level author, but not the source element. As requested, I will create a separate MR for it. Hence, this one here just partially fixes #2256.

Three things to note upfront:

1. I'm not entirely sold on the method name `parse_and_update_author`. Maybe sb. comes up with something better.
2. Not sure if I should have created a new Atom feed for the test case or whether it's alright to reuse the already existing one.
3. Lastly, I don't consider myself an experienced C++ programmer, so please have a thorough look.

Minoru reviewed (changes requested):
Thanks! This looks great.

`parse_and_update_author` seems fine to me. `extract_author` would be shorter, but I don't think it would have made the code any more clearer.

Re-using the existing Atom file seems fine too, however I'd prefer to have a separate test for this feature. That way, if we accidentally break it later, it'd be immediately obvious *what* got broken. Can you please add it?

der-lyse commented:
Yes, good idea. I added a new test feed with even multiple feed authors to also test the string concatenation. In fact this test feed is illegal, since all `<author>`s must appear before all `<entry>` elements according to the W3C Feed Validator. But in order to be fault-tolerant, I decided to manifest this in a test.

The fourth entry with explicit empty authors also cements an implementation decision, a reasonable one, though, in my opinion.

I rebased onto current master and just added the new test case at the bottom and the new feed file. Nothing else was changed. `extract_author` seems like a good name, too, but the `extract` makes me think it returns something, so I kept the old name.

Minoru reviewed (approved):
Excellent, thank you very much!
